### ISSUE TITLE
package-comments flags main as of 1.2.2

### ISSUE THREAD

**Issue by twmb** [mgechev/revive#740]
This might not be a bug, but I'd like to clarify if this is intentional. It's not common to document main packages, I personally plan to disable the rule if it continues to lint documenting main.

**Comment by chavacava** [comment]
Hi @twmb, thanks for filling the issue.

Well, even if documenting `main` packages might be not so common, the [official guideline for documenting GO code](https://tip.golang.org/doc/comment) says they should be documented:

>**Every** package should have a package comment introducing the package. 

and it also provides an example for a `main` package:
```go
/*
Gofmt formats Go programs.
It uses tabs for indentation and blanks for alignment.
Alignment assumes that an editor is using a fixed-width font.
[...]
When gofmt reads from standard input, [...]
formatted by piping them through gofmt.
*/
package main
```

As a reference, [standard GO tools](https://cs.opensource.google/go/x/tools/+/master:cmd/) have documented main packages.

Must `revive` lint main package documentation? I think it must but I'm open to discuss about.

**Comment by hawkingrei** [comment]
We have meet this problem, too. but we are not main package.

https://github.com/pingcap/tidb/pull/37413

https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_build/detail/tidb_ghpr_build/78196/pipeline

```
util/texttree/texttree.go:15:1: package-comments: should have a package comment (revive)
```

https://github.com/pingcap/tidb/blob/a80078538d366d55acfc34dd772b1977a720a9bc/util/texttree/texttree.go#L14-L16

**Comment by twmb** [comment]
> Must `revive` lint main package documentation?

I'm open to it, but I think it's worth a quick survey of the packages that already use revive to see if they currently document main. I never have -- I document my main packages elsewhere. As well, main packages do not follow the common `Package main ...`, as seen in the linked post above.

**Comment by chavacava** [comment]
> We have meet this problem, too. but we are not main package.
> 
> util/texttree/texttree.go:15:1: package-comments: should have a package comment (revive)
> 
> https://github.com/pingcap/tidb/blob/a80078538d366d55acfc34dd772b1977a720a9bc/util/texttree/texttree.go#L14-L16

Hi @hawkingrei , I've checked the [code you point in your comment](https://github.com/pingcap/tidb/tree/a80078538d366d55acfc34dd772b1977a720a9bc/util/texttree) and I've seen there is no documentation for the `texttree` package. Thus `revive` pointing that lack of documentation seems OK to me. Do you agree or I'm missing something?

**Comment by chavacava** [comment]
> As well, main packages do not follow the common `Package main ...`, as seen in the linked post above.

Yes, [`revive` takes that into account](https://github.com/mgechev/revive/blob/553604eaced55ff3b0262525fa1cc19233cce297/rule/package-comments.go#L153)

**Comment by mcandre** [comment]
A package level API doc for a main package may seem like overkill.

However, projects that provide multiple applications can use the separate docs to explain the purpose of each application.

For example:

```go
// Package main implements a GIF editor application.
package main
```

Even a Go project with only one application, can still use this API doc to encourage the author to say a few words about the purpose of their application. There are tons of projects, FOSS and enterprise, with zero explanation for their tools are intended to accomplish.

**Comment by alexandear** [comment]
Answered.
